### PR TITLE
simplified for loop

### PR DESCRIPTION
**Pull request by MCWertGaming**
As introduced in #64 and implemented in #68: This pull-request tries to simplify the get_cursor_position() function. The code should be checked of logic mistakes. I have tried to just change the code, but don't touch the logic at all. I hope it's still intakt!

### CODE REVIEW

**Comment by certik** [comment]
The fact that even you forgot to append `std::flush` and had to fix it in 4bb0f822d10f91c7dd9392d198e629df08d9b87d is an argument against #81.

**Comment by MCWertGaming** [comment]
the fix in 4bb0f82 was due to a misstake in the old pull-request #68, I made the misstake there. I'll have to resolve that conflict in git, but will do that, after #81 was merged of closed. Like I said both is absolutely fine! I just did brain dead copy paste here, sorry for that!
I'll change it to the write() function, if you decide to drop #81.

**Comment by MCWertGaming** [comment]
Because we dropped #81 and changed it to #87, I'll wait for #87 to get merged and then change the std::cout here back to the write function. You can take a look at the logic then @certik.

**Comment by certik** [comment]
#87 is merged, you can go ahead and finish the PR now.

**Comment by MCWertGaming** [comment]
done!

File: cpp-terminal/terminal.h
Comment (certik) [inline cpp-terminal/terminal.h:384]:
```suggestion
    void get_cursor_position(int& rows, int& cols) const
```

File: cpp-terminal/terminal.h
Comment (certik) [inline cpp-terminal/terminal.h:388]:
```suggestion
        for (unsigned int i = 0; i < sizeof(buf) - 1; i++) {
```

File: cpp-terminal/terminal.h
Comment (certik) [inline cpp-terminal/terminal.h:391]:
```suggestion
            if (buf[i] == 'R') {
```

File: cpp-terminal/terminal.h
Comment (certik) [inline cpp-terminal/terminal.h:395]:
```suggestion
                if (i < 5) {
                    throw std::runtime_error("get_cursor_position(): too short response");
                } else {
                    buf[i] = '\0';
                }
```

File: cpp-terminal/terminal.h
Comment (certik) [inline cpp-terminal/terminal.h:408]: When you have multineline if / while statements, use `{}` to prevent a mistake in the future:
```suggestion
            if (buf[i] == '\0') {
                break;
            }
```
Or put them on the same line:
```suggestion
            if (buf[i] == '\0') break;
```

**Review by certik — CHANGES_REQUESTED** [review]
I think it looks good. It seems it is equivalent. I left some minor formatting suggestions.

**Comment by MCWertGaming** [comment]
about the formating: the code base has no real coding style. Sometimes I have seen 2 spaces, sometimes 4 spaces, sometimes a new line after curly brackets, sometimes there is none. We should really use clang format for that. Then there wont be such things to check anymore.

File: cpp-terminal/terminal.h
Comment (MCWertGaming) [reply cpp-terminal/terminal.h:408]: about the {}, that's true, i sometimes forget that. There is a compiler waning on that btw.

**Review by certik — APPROVED** [review]
I think this looks good.

**Pull request merged**
